### PR TITLE
Invite Notification

### PR DESCRIPTION
**Pull request by cp-pratik-k**
### Purpose
 - User shoud be able to get notification if its invited on space.
 - Hr should be able to get notification about invite accept.
 
### Summary of Changes
 - Add implementation for invite notification service
 - Add implementation for send invitation notification to user
 - Add implementation for send invitation accept notification to hr.

### Test steps
 - Run flutter test

### Conformity

- [x] Followed [git guidelines](https://github.com/canopas/flutter-developer-roadmap/blob/main/GitGuideline.md) for creating commit messages and Pull Request guidelines.
- [x] Self-approved the PR - reviewed the PR as a reviewer and gave it self-approval if everything is ok. If not, made the required changes.
- [x] Ensured that the PR satisfies all specified requirements in the ticket, including bug fixes and new features.
- [x] Provided test steps, including steps to reproduce the issue or test the new functionality, ensuring other team members can verify the changes.
- [x] Added/Updated proper code comments to make it easy-to-understand for other developers.
- [x] Reused code (if the same code was written twice, made it common and reused it at both places).
- [x] Removed unused or commented code if not required.
- [x] Ensured proper Dart naming conventions were used for variables, classes, and methods.
- [x] Localized user-facing strings.
- [x] Included screenshots/videos of behavior changes: Provided visual evidence of any changes to UI or behavior for easier review and understanding in the PR description.
- [x] Implemented proper error handling: Ensured that the code anticipated and handled potential errors and edge cases gracefully.
- [x] Avoided introducing technical debt: If the PR introduces technical debt, created and linked appropriate tickets for future resolution.
- [x] Included relevant unit tests: Wrote unit tests that focused on testing behavior and functionality, rather than merely covering lines of code.
- [x] Ensured code was performant and scalable: Verified that the changes did not introduce performance issues or bottlenecks and could scale as needed.
- [x] Ensured comments were up-to-date and relevant to the code to describe complex logic and to add understanding for other developers.
- [x] Marked the PR as ready before submitting it for review.

### CODE REVIEW

**Review by cp-sneha-s — APPROVED** [review]

File: lib/data/services/mail_notification_service.dart [hunk 1]
```diff
@@ -71,4 +71,36 @@ class NotificationService {
...
+          await httpClient.post(Uri.https(baseURL, '/api/acceptence'),
+              body: json.encode({
+                "receiver": receiver,
+                "sender": sender,
+              }));
+      return response.statusCode == 200;
+    } on Exception {
+      return false;
+    }
+  }
```
Comment (cp-sneha-s) [inline lib/data/services/mail_notification_service.dart:105]: I think we have to add a log in case of exception.

Comment (cp-sneha-s) [reply lib/data/services/mail_notification_service.dart]: Let's add log like: "Invitation notification mail send successfully" in case of success

Comment (cp-pratik-k) [reply lib/data/services/mail_notification_service.dart]: i don't think we have to add log in firebase crashlytics in case of success.

**Review by cp-sneha-s — APPROVED** [review]
